server/controllers/profile.js

```diff
@@ -1,56 +1,94 @@
+  const {
+    firstName,
+    lastName,
+    gender,
+    email,
+    phoneNumber,
+    address,
+    description,
+  } = req.body;
+  if (firstName && typeof firstName !== 'string') {
+    res.status(404);
+    throw new Error('Provide input as a String for firstName');
+  }
+  if (lastName && typeof lastName !== 'string') {
+    res.status(404);
+    throw new Error('Provide input as a String for lastName');
+  }
+  if (gender && typeof gender !== 'string') {
+    res.status(404);
+    throw new Error('Provide input as a String for gender');
+  }
+  if (email && typeof email !== 'string') {
     res.status(404);
-    throw new Error("Provide input as a String for firstName");
+    throw new Error('Provide input as a String for email');
   }
-  if (lastName && typeof lastName !== "string") {
+  if (phoneNumber && typeof phoneNumber !== 'string') {
     res.status(404);
-    throw new Error("Provide input as a String for lastName");
+    throw new Error('Provide input as a String for phone number');
   }
-  if (description && typeof description !== "string") {
+  if (address && typeof address !== 'string') {
     res.status(404);
-    throw new Error("Provide input as a String for description");
+    throw new Error('Provide input as a String for address');
   }
-  if (availability && typeof availability !== "boolean") {
+  if (description && typeof description !== 'string') {
     res.status(404);
-    throw new Error("Provide input as a Boolean for availability");
+    throw new Error('Provide input as a String for description');
   }
 
-  let newProfile = new profile(req.body);
-  newProfile.save(function (err, result) {
+  const newProfile = new Profile(req.body);
+  newProfile.save(function(err, result) {
     if (err) res.send(err);
     res.json(result);
   });
 };
 
-const getProfile = (req, res) => {
-  profile.findOne({ _id: req.params.profileID }, function (err, result) {
-    if (err) res.send(err);
-    res.json(result);
+const getProfile = asyncHandler(async (req, res, next) => {
+  const profile = await Profile.findById(req.params.profileID);
+
+  if (!profile) {
+    res.status(401);
+    throw new Error('Not authorized');
+  }
+
+  res.status(200).json({
+    success: {
+      profile: {
+        id: profile._id,
+        firstName: profile.firstName,
+        lastName: profile.lastName,
+        gender: profile.gender,
+        email: profile.email,
+        phoneNumber: profile.phoneNumber,
+        address: profile.address,
+        description: profile.description,
+      },
+    },
   });
-};
+});
 
-const updateProfile = (req, res) => {
-  profile.updateOne(
-    { _id: req.query.profileID },
+const updateProfile = asyncHandler(async (req, res, next) => {
+  Profile.findByIdAndUpdate(
+    req.body.id,
     req.body,
-    function (err, result) {
-      if (err) res.send(err);
-      res.json(result);
+    { new: true },
+    (err, todo) => {
+      if (err) return res.status(500).send(err);
+      return res.send(todo);
```

Comment:
Since you're already using an `async` function, it would be good to be consistent with `async/await` instead of using this callback.